### DISCUSSION TITLE
Lots of 'context canceled' error messages in 'READ' Components

### DISCUSSION THREAD

**Discussion by pingping95** [grafana/mimir#5555] [Help and support]
Hi Mimir teams! Thanks for developing a good monitoring tool.

I've already checked all of the Mimir Slack Channel, Github Issue, and Github Discussion,

but I still haven't solved it.

I tried to solve it on my own for a few days, but I couldn't. So I ask for help.

<img width="1753" alt="image" src="https://github.com/grafana/mimir/assets/67780144/bb0f3005-6146-4521-908e-c65c3c4da77e">

1. `Querier, Store-Gateway` is experiencing a number of gRPC errors.
I don't know if it's a normal log or if it's actually a problematic log.
May I know if this is a problem?
If there is a problem, I wonder what kind of settings can be used to solve it.

```
err="rpc error: code = Canceled desc = context canceled
```
<br>

<img width="1206" alt="image" src="https://github.com/grafana/mimir/assets/67780144/9f2169f4-bbf9-4e12-acd5-ee90bdb7f12b">

<br>
<br>

2. `Distributor , Ingester` is experiencinig lots of errors like this.
We are using Prometheus Agent for remoteWriting samples to Mimir.
I don't have enough knowledge on the time series side, so I don't know how to solve these redundant data issues,
Anyone can help me?

```
the sample has been rejected because another sample with the same timestamp, but a different value, has already been ingested
```

**Comment by dimitarvdimitrov** [comment]
> 1. Querier, Store-Gateway is experiencing a number of gRPC errors.

if those correlate with 4xx responses to queries, it's likely that the queries are exceeding some limits or contain invalid queries. These queries aren't evaluated to completion. Instead Mimir returns a 4xx error and cancels the query execution and you see the context cancelled logs. You can check the responses in the query-frontend logs to help find what's broken. A logQL query to find those is this

```
{job="mimir/query-frontend"} |= "query stats" | logfmt | status!="success"
```

> 2. Distributor , Ingester is experiencinig lots of errors like this

I don't know off the top of my head. Is it possible that there are colliding metrics from multiple targets that the prometheis are scraping? For example, if you drop the `hostname` label from a timeseries, then different series from multiple hosts will look like the same timeseries and Mimir will reject one of them.
